src/tofromdatavalues.jl

```diff
@@ -93,7 +93,15 @@ IteratorInterfaceExtensions.getiterator(x::MyTable) = Tables.datavaluerows(x)
 ```
 """
 function datavaluerows(x)
-    r = Tables.rows(x)
+    x2 = if Tables.columnaccess(x)
+        Tables.columntable(x)
+    elseif Tables.rowaccess(x)
+        x
+    else
+        error("Invalid input table, must either declare to have a Tables.rows or Tables.columns method.")
+    end
+
+    r = Tables.rows(x2)
```

Comment:
```suggestion
    r = Tables.rows(Tables.columnaccess(x) ? Tables.columntable(x) : x)
```
Hmmmm.....yeah, I think this is probably fine to change; and it shouldn't affect DataFrames because calling `columntable(::ColumnTable)` is a no-op.

The reason for not throwing an error in my suggestion is because someone may have, say, a `Vector{Any}` where elements are NamedTuples with `missing` values, but they want to use that in Query.jl. A `Vector{Any}` doesn't have `rowaccess` defined, but it will work with `Tables.rows`, assuming it actually has valid "rows" as elements.

Reply:
Yep, that makes sense!